### PR TITLE
feat: implement Eq for Ipld

### PR DESCRIPTION
**Pull request by vmx**
Implement the `Eq` trait for the `Ipld` enum. NaN floats are never equal to another. Though in the Ipld Data model they are forbidden. This means that we can implement `Eq` for it.

As it's currently not strictly enforced that an `Ipld::Float` does not contain a NaN, we make at least sure that in case we would encounter it, that it is treated as equal.

For more information about the dicussion about this topic, see https://github.com/ipld/libipld/issues/171.

### CODE REVIEW

File: src/ipld.rs [hunk 1]
```diff
@@ -363,4 +388,12 @@ mod tests {
         let ipld = Ipld::Map(map);
         assert_eq!(ipld.get("a").unwrap(), Some(&Ipld::Integer(0)));
     }
+
+    // NaN floats are forbidden in the IPLD Data Model, but still make sure they are treated as
+    // equal in case they accidentally end up there.
+    #[test]
+    fn test_partial_eq_nan() {
```
Comment (rvagg) [inline src/ipld.rs:395]: What happens when you compare different nans? Try a 3-way comparison between `f64::NAN` with `f64::from_ne_bytes([0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0xf0, 0x7f])` and `f64::from_ne_bytes([0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x7f])`, they're all going to be "equal" aren't they? How much do we care about such things? (Also one of the reasons that NaN is forbidden).

Comment (Stebalien) [reply src/ipld.rs:395]: IMO, it's okay to define these as equal. Especially given that we'll reject all these when we serialize and deserialize them.

**Review by Stebalien — APPROVED** [review]
The code looks good to me, but I feel more comfortable if you could convince someone who didn't propose this, like @rvagg.

**Comment by rvagg** [comment]
whatever, 🙈 🙉

**Pull request merged**